Tabs are working properly with vector icons

cmann50 opened the pull request:
- Full screen map displays
- Clicking icons shows and alert
- Cleanup code to match standards

cmann50 committed:
Add tabs to app

ericmasiello commented on src/TrailMap.ios.js:
```diff
@@ -0,0 +1,120 @@
+'use strict';
```
So does his mean we now have two versions of TrailMap?

cmann50 replied:
Unless I messed something up, it should just be your code from develop with the tabs in it.  This is a pr from a feature branch to develop, so it should merge down.  I did a trial run to make sure it does a Ff merge.  If it looks off let me know and I'll fix it

cmann50 replied:
Actually, let me see if I can cleanup the commits with a squash or rebase

cmann50 commented:
Ok, it should be better now.  I see what you meant by duplicate TrailMap.  I think it must have been a local copy that got added even though I thought I deleted it.

Either way, I want through and did an interactive rebase and squashed all those commits into a single commit, so it should be a lot easier to review.

ericmasiello commented on src/components/trail.style.js:
```diff
@@ -1,11 +1,43 @@
+  mapView: {
+      flex: 1,
+      
+      top: 0,
```
i don't think you ned the top, left, bottom, and right values here. try removing them and see if they make any difference. typically those are only used with absolute or relative position elements

cmann50 replied:
You are right.  Removed locally.

ericmasiello replied:
Yeah i Just confirmed, You can remove all the top, bottom, left, and right properties on these rules. Jut keep flex and height props.

cmann50 replied:
These should be removed.  I re-pushed with an amended commit.

> On Feb 15, 2016, at 5:34 PM, ericmasiello notifications@github.com wrote:
> 
> In src/components/trail.style.js https://github.com/ericmasiello/Trail-Conservancy-Mobile/pull/5#discussion_r52949965:
> 
> > @@ -1,11 +1,43 @@
> >  export default {
> >    container: {
> > -    flex: 1
> > -    flex: 1,
> > -  },
> > -  mapView: {
> > -      flex: 1,
> > -   
> > -      top: 0,
> >   Yeah i Just confirmed, You can remove all the top, bottom, left, and right properties on these rules. Jut keep flex and height props.
> 
> —
> Reply to this email directly or view it on GitHub https://github.com/ericmasiello/Trail-Conservancy-Mobile/pull/5/files#r52949965.

ericmasiello commented:
Cool thanks for doing that. can you explain what a rebase actually does? I still don't get it.

ericmasiello commented:
Also, I'll take a look at the acutal working code tomororw

cmann50 commented:
I should show it to you using my windows 95 looking git tool.

Basically, it allows you to a branch and make it's history linear.  So I had merged in two branches into a local copy of /develop (it was easier to do a 3 way compare and pull the code I wanted), then branched that into a new branch feature/final-tab.  The I submitted a pull request.

The history was really convoluted, so I did a rebase.  This first command ' git merge-base feature/tab-final develop'  gives you the commit that is where I branched from develop (into all these various merges etc).  From the root commit, I ran a rebase.  The rebase walked up each commit and combined it into the develop branch.  So at that point all the commits are now on develop in a linear history.

Every commit i processed, I had to jump over to my git tool to resolve conflicts, then run 'git rebase --continue'  again.

Once complete, I was good to go.

I also squashed all the commits into a single commit, so not only were the linear in the history, but they were turned into the 'Add tabs to app' commit.

Apparently it's very common to do a rebase w/ squash before requesting a pull because it means there's the minimal amount of files to review (since it's now only a single commit).

 git merge-base feature/tab-final develop
 git rebase -i  49874f0ac82fec8f358f0d84f31a418ef5bb351d
 git rebase --continue
 git rebase --continue
 git rebase --continue
 git rebase --continue
 git rebase --continue
 git rebase --continue
 git rebase --continue

> On Feb 14, 2016, at 5:38 PM, ericmasiello notifications@github.com wrote:
> 
> Cool thanks for doing that. can you explain what a rebase actually does? I still don't get it.
> 
> —
> Reply to this email directly or view it on GitHub https://github.com/ericmasiello/Trail-Conservancy-Mobile/pull/5#issuecomment-183994877.

ericmasiello commented on src/components/menu.component.js:
```diff
@@ -0,0 +1,108 @@
+import Icon from 'react-native-vector-icons/Ionicons';
+
+const {
+  Component,
```
These can be put on the same line as the import React statement.

e.g. 

import React, { Component, NavigatorIOS } from ' react-native'

cmann50 replied:
fixed locally

ericmasiello commented on src/components/menu.component.js:
```diff
@@ -0,0 +1,108 @@
+    };
+  }
+
+  renderSection(route, ref) {
```
where is this function used?

cmann50 replied:
not uses, removed locally

cmann50 commented:
Re-pushed with changes (it should auto-update on the PR in theory)

cmann50 committed:
fix requested issues

ericmasiello commented on src/components/menu.component.js:
```diff
@@ -0,0 +1,87 @@
+
+import Icon from 'react-native-vector-icons/Ionicons';
+
+import React, {Component, NavigatorIOS, TabBarIOS, View, AlertIOS,} from 'react-native';
```
Cool! I think NavigatorIOS can be removed too

ericmasiello replied:
Just fix that and you can merge it into develop.

Btw, I configured this repo so you can't pushed directly to master or develop. So any merges into develop will need to be done via pull requests.

cmann50 replied:
That's smart since I accidentally pushed to develop yesterday and it saved my ass.

cmann50 commented:
Note that there's a bunch of stuff caught by the linter that I plan to cleanup in the next push.
